src/solve/multiple_shooting.jl

```diff
@@ -236,7 +231,7 @@ function __solve(prob::BVProblem, _alg::MultipleShooting; odesolve_kwargs = (;),
                 nodes); resid_prototype, jac = jac_fn, jac_prototype)
         nlprob = NonlinearProblem(loss_function!, u_at_nodes, prob.p)
         sol_nlsolve = __solve(nlprob, alg.nlsolve; nlsolve_kwargs..., verbose, kwargs...)
-        # u_at_nodes = sol_nlsolve.u
+        u_at_nodes = sol_nlsolve.u::typeof(u0)
```

Comment:
why is the type assert needed?

Reply:
I tried figuring out where the exact type instability is coming from, but Cthulhu also breaks down...